docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,328 @@
+This attribute doesn't have very well-defined semantics.
+
+Type bridging is not done, so the parameter and return types should
+correspond directly to types accessible in C.
+In most cases, it is preferable to define a static method on an `@objc`
+class instead of using `@_cdecl`.
+
+For potential ideas on stabilization, see
+[Formalizing `@cdecl`](https://forums.swift.org/t/formalizing-cdecl/40677).
+
+## `@_disfavoredOverload`
+
+Marks a declaration that the type checker should try to avoid using. When the
+expression type checker is considering overloads, it will prefer a solution
+with fewer `@_disfavoredOverload` declarations over one with more of them.
+
+Historically, it was designed to help with `ExpressibleByXYZLiteral` types.
+The type checker strongly prefers to give literals their default type
+(e.g. `Int` for `ExpressibleByIntegerLiteral`,
+`String` for `ExpressibleByStringLiteral`, etc.).
+If an API should prefer some other type, but accept the default too,
+marking the declaration taking the default type with `@_disfavoredOverload`
+gives the desired behavior:
+
+```swift
+extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
+
+extension Text {
+  // We want `Text("foo")` to use this initializer:
+  init(_ key: LocalizedStringKey) { ... }
+
+  // But without @_disfavoredOverload, it would use this one instead,
+  // because that lets it give the literal its default type:
+  @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
+}
+```
+
+`@_disfavoredOverload` is a stop-gap design; eventually the type-checker
+behavior should be fixed and the attribute should be removed/made into a no-op.
+
+Its main use case is to work around clearly undesirable overload resolution,
+not to arbitrarily manipulate overload priority.
+
+## `@_dynamicReplacement(for:)`
+
+Marks a function as the dynamic replacement for another `dynamic` function.
+This is similar to method swizzling in other languages such as Objective-C,
+except that the replacement happens at program start (or loading a shared
+library), instead of at an arbitrary point in time.
+
+For more details, see [Dynamic method replacement](https://forums.swift.org/t/dynamic-method-replacement/16619).
+
+## `@_distributedActorIndependent`
+
+## `@_effects`
+
+## `@_exported`
+
+Use to mark an imported module to re-export all its declarations.
+This attribute is most commonly used by overlays.
+
+```swift
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+func g() {
+  N.f() // OK
+}
+```
+
+## `@_fixed_layout`
+
+## `@_hasInitialValue`
+
+## `@_hasMissingDesignatedInitializers`
+
+## `@_hasStorage`
+
+## `@_implementationOnly`
+
+Used to mark an imported module as an implementation detail.
+This prevents types from that module being exposed in API
+(types of public functions, constraints in public extension etc.)
+and ABI (usage in `@inlinable` code).
+
+## `@_implements`
```

Comment:
Proposed definition:

An attribute that indicates that a function with one name satisfies a protocol requirement with a different name. This is especially useful when two protocols declare a requirement with the same name, but the conforming type wishes to offer two separate implementations.

```
protocol P { func foo() }

protocol Q { func foo() }

struct S : P, Q {
  @_implements(P, foo())
  func foo_p() {}
  @_implements(Q, foo())
  func foo_q() {}
}
```